program/src/processor/initialize_stake.rs

```diff
@@ -14,8 +23,97 @@ use crate::instruction::accounts::{Context, InitializeStakeAccounts};
 /// 2. `[]` Validator vote account
 /// 3. `[]` System program
 pub fn process_initialize_stake(
-    _program_id: &Pubkey,
-    _ctx: Context<InitializeStakeAccounts>,
+    program_id: &Pubkey,
+    ctx: Context<InitializeStakeAccounts>,
 ) -> ProgramResult {
+    // Accounts validation.
+
+    // 1. config
+    // - owner must the stake program
+    // - must be initialized
+
+    require!(
+        ctx.accounts.config.owner == program_id,
+        ProgramError::InvalidAccountOwner,
+        "config"
+    );
+
+    let data = &ctx.accounts.config.try_borrow_data()?;
+
+    let config = bytemuck::try_from_bytes::<Config>(data)
+        .map_err(|_error| ProgramError::InvalidAccountData)?;
+
+    require!(
+        config.is_initialized(),
+        ProgramError::UninitializedAccount,
+        "config"
+    );
+
+    // 2. validator_vote
+    // - owner must be the vote program
+    // - must be initialized
```

Comment:
The numbering is not associated with the account order, it represents the steps on the validation. The reason the `validator_vote` comes first is because the we need the validator pubkey (inside the vote account) to validate the derivation of the stake account.

Perhaps we can remove the numbering if that is confusing with the account order.